src/main/java/gov/cms/madie/services/QrdaService.java

```diff
@@ -0,0 +1,88 @@
+package gov.cms.madie.services;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gov.cms.madie.Exceptions.QrdaServiceException;
+import gov.cms.madie.dto.QRDADto;
+import gov.cms.madie.dto.SourceDataCriteria;
+import gov.cms.madie.models.dto.TranslatedLibrary;
+import gov.cms.madie.models.measure.QdmMeasure;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class QrdaService {
+
+  private final CqmMeasureMapper mapper;
+  private final TranslationServiceClient translationServiceClient;
+  private final QrdaClient client;
+  private final ObjectMapper objectMapper;
+
+  public List<String> generateQrda(QdmMeasure measure, String accessToken) {
+    // get Libraries
+    List<TranslatedLibrary> translatedLibraries =
+        translationServiceClient.getTranslatedLibraries(measure.getCql(), accessToken);
+    List<String> elms =
+        translatedLibraries.stream()
+            .map(translatedLibrary -> translatedLibrary.getElmJson())
+            .collect(Collectors.toList());
+
+    List<SourceDataCriteria> dataCriteria =
+        translationServiceClient.getRelevantDataElements(measure, accessToken);
+
+    QRDADto dto = null;
+    try {
+
+      dto =
+          QRDADto.builder()
+              .measure(objectMapper.writeValueAsString(mapper.measureToCqmMeasure(measure, elms)))
+              .testCases(measure.getTestCases())
+              .sourceDataCriteria(dataCriteria)
+              .options(buildOptions(measure))
+              .build();
+    } catch (JsonProcessingException e) {
+      throw new QrdaServiceException("Problem mapping the measure for QRDA generation", e);
+    }
+
+    // send to qrda
+    return client.getQRDA(dto, accessToken);
+  }
+
+  private Map<String, Object> buildOptions(QdmMeasure measure) {
+    Long startTime =
+        Long.parseLong(
+            measure
+                .getMeasurementPeriodStart()
+                .toInstant()
+                .atZone(ZoneId.systemDefault())
```

Comment:
Need to remember to verify the timezone used in prod.

Reply:
I thought since it was handling all of this part on server side it wouldn't matter for conversion sake especially since it ends up not using the zone for the final string anyway

Reply:
I think this might cause issues because, I believe, we're trying to force measurement period to UTC in the browser, and then this will convert that to local/default timezone. Which, if we are not explicitly setting to UTC on the server, would probably be America/New_York since we run in AWS US-East1 region.
This means what we might see is for measurement period start, in the browser they set `01/01/2024`, which I'm seeing in the JSON payload as `2024-01-01T00:00:00.000+00:00`.
Then on the backend, this would be turned into a ZonedDateTime `2023-12-31T19:00-05:00[America/New_York]` and LocalDateTime of `2023-12-31T19:00`.

Reply:
And this might apply to measurementPeriodEnd logic below, as well as the CqmMeasureMapper logic

Reply:
@chubert-sb - resolving this for purposes of merge, but let's follow up on this with some dedicated testing and make sure it outputs properly in the export